src/ModelRegion.cpp

```diff
@@ -427,11 +436,80 @@ namespace geopm
         }
     }
 
+    TimedStreamModelRegion::TimedStreamModelRegion(double big_o_in,
+                                                   int verbosity,
+                                                   bool do_imbalance,
+                                                   bool do_progress,
+                                                   bool do_unmarked)
+        : StreamModelRegion(big_o_in, verbosity, do_imbalance, do_progress, do_unmarked)
+    {
+        m_name = "timed_stream";
+
+        if (big_o_in > 10.0) {
+            /// @todo: adjust to reuse same buffer when big-o would exhaust system memory. This depends on number of ranks.
+            std::cerr << "Warning: <geopm> timed_stream region may run out of memory for large big-o values." << std::endl;
+        }
+
+        if (m_verbosity) {
+            std::cout << "Calibrating timed_stream region to " << big_o_in << " seconds.  Please wait..." << std::endl;
+        }
+        double measured_time = 0;
+        bool done = false;
+        double new_big_o = big_o_in;
+        double lower = 0.0;
+        const double EPS = big_o_in / 100.0;
+        const int MAX_ITERATIONS = 20;
+        int iteration_count = 0;
+        double step_size = new_big_o;
```

Comment:
It looks like the intention of the algorithm w.r.t. evaluating the search space is to:
1. Halve the the step size if it ran too long.
2. If it ran too short, add the step size to the big O and re-run.
a. If it was too short on the first eval, big O is doubled.  Otherwise, big-O is set to <= 0.5 `big_o_in`.

I just want to be sure I'm groking that right.  A comment would be nice.

Reply:
Yes that's correct.  I'm not 100% sure if this is the best algorithm. The original version was increasing the big-o by (0.5 * (upper-lower)) but this is simpler and seems to converge faster when the initial big-o is too small. I haven't done very many experiments with different big-o values and OMP_NUM_THREADS combinations.